### PR TITLE
[AD-70] on authenticate do not support blank password

### PR DESCRIPTION
**Pull request by andrea-patricelli**

### CODE REVIEW

File: src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 /**
  * Copyright (C) 2011 ConnId (connid-dev@googlegroups.com)
- *
+ * <p>
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java:3]: Revert such changes in the license header

File: src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java [hunk 2]
```diff
@@ -140,4 +139,18 @@ public Uid resolveUsername() {
         }
         return authnObject.getUid();
     }
+
+    private static boolean isNotBlankCredentials(final GuardedString credentials) {
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java:143]: With such approach you are basically duplicating the same `GuardedString` access performed elsewhere in the same class: can't we do that once, instead?

**Review by ilgrosso — CHANGES_REQUESTED** [review]

File: src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java (hunk 2 above)
Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java:143]: Yes, the other access was too "inner" so I opted for this solution. Here https://github.com/Tirasa/ConnIdADBundle/blob/master/src/main/java/net/tirasa/connid/bundles/ad/ADConnection.java#L204-L213

But I think that is doable. let me work on it.

File: src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java [hunk 3]
```diff
@@ -140,4 +140,4 @@ public Uid resolveUsername() {
         }
         return authnObject.getUid();
     }
-}
+}
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java:143]: Please revert this chage.

**Review by ilgrosso — CHANGES_REQUESTED** [review]

File: src/main/java/net/tirasa/connid/bundles/ad/ADConnection.java [hunk 4]
```diff
@@ -206,6 +206,9 @@ private LdapContext connect(String principal, GuardedString credentials) {
 
                     @Override
                     public void access(char[] clearChars) {
+                        if(clearChars == null || clearChars.length == 0){
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/ad/ADConnection.java:209]:
```suggestion
                        if (StringUtil.isBlank(clearChars)) {
```

**Review by ilgrosso — CHANGES_REQUESTED** [review]

File: src/main/java/net/tirasa/connid/bundles/ad/ADConnection.java (hunk 4 above)
Comment (ilgrosso) [reply src/main/java/net/tirasa/connid/bundles/ad/ADConnection.java:209]: Please ignore this comment

**Review by ilgrosso — APPROVED** [review]

**Pull request merged**